core/src/main/java/io/aiven/klaw/service/BaseOverviewService.java

```diff
@@ -224,9 +215,19 @@ protected List<AclOverviewInfo> applyFiltersAclsForSOT(
 
     List<AclOverviewInfo> aclList = new ArrayList<>();
     AclOverviewInfo mp;
-
+    Set<Integer> set =
+        aclRequestsRepo
+            .findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
+                tenantId,
+                aclsFromSOT.get(0).getEnvironment(),
+                RequestStatus.DELETED.value,
+                aclsFromSOT.get(0).getTopicname())
+            .stream()
+            .map(AclRequests::otherParams)
+            .collect(Collectors.toSet());
     for (Acl aclSotItem : aclsFromSOT) {
```

Comment:
the map function needs to use "getOtherParams" instead of "otherParams" that will then return a String so in the end it will end up looking like this.

   Set<String> set =
        aclRequestsRepo
            .findAllByTenantIdAndEnvironmentAndRequestStatusAndTopicname(
                tenantId,
                aclsFromSOT.get(0).getEnvironment(),
                RequestStatus.DELETED.value,
                aclsFromSOT.get(0).getTopicname())
            .stream()
            .map(AclRequests::getOtherParams)
            .collect(Collectors.toSet());
    for (Acl aclSotItem : aclsFromSOT) {
      if (aclSotItem.getAclip() != null || aclSotItem.getAclssl() != null) {
        boolean deleteRequestExists = set.contains(Integer.valueOf(aclSotItem.getReq_no()));